Khan/wonder-blocks
# WB-1652.2: Add theming support to ModalPanel component

## Summary:

Following up on #2140, this PR focuses on adding theming support to the
ModalPanel component. This PR also includes a new theme for the `khanmigo` theme
that is used to support the `eggplant` background.

Issue: WB-1652

## Test plan:

1. Verify that the ModalPanel component can be themed by using the `khanmigo` theme.

http://localhost:6061/?path=/story/modal-building-blocks-modalpanel--dark

Check that the background color of the modal panel is `eggplant` when using the
`khanmigo` theme and is `darkBlue` when using the default theme.


https://github.com/Khan/wonder-blocks/assets/843075/3ebee616-6a23-42a2-aee1-6b7f15faba6c

2. Verify that the new ModalDialog story displays correctly the new `khanmigo`
theme.
<img width="772" alt="Screenshot 2023-12-13 at 6 23 14 PM" src="https://github.com/Khan/wonder-blocks/assets/843075/2d48e30f-4f28-4ba3-94c4-b2d52f2bea95">
